controllers/argocd/argocd_controller.go

```diff
@@ -119,6 +121,66 @@ func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		}
 	}
 
+	// update image and version
+	if policy, ok := obj.Annotations[constants.AnnotationImageVersionUpdatePolicy]; ok && policy != constants.ImageVersionUpdatePolicyNone {
+		original := obj.Spec.DeepCopy()
+
+		// set the ArgoCD image and version
+		if (policy == constants.ImageVersionUpdatePolicyAlways) ||
```

Comment:
I tried but didn't came up with an easier solution, it is difficult because there aren't any common getters and setters for the image and version across ArgoCD, Dex and Redis